template/workflow/envs/environment.yaml.jinja

```
@@ -0,0 +1,20 @@
+name: {{project_slug}}
+channels:
+  - conda-forge
+dependencies:
+  - python={{python_version}}
+  - pip
+  - pyflakes
+  - numpy
+  - scipy
+  - matplotlib
+  - ipython
+  - pip:
+    - snakemake
+    - ipykernel
+    - ipympl
+    - numpy
+    - pyfar
+    - sofar
+    - pyrato
```

```suggestion
  - ipython
  - ipykernel
  - ipympl
  - pip:
    - snakemake
    - pyfar
    - sofar
    - pyrato
```
If possible, install everything available from conda to avoid issues with dependency resolution

---

Conda will always install conda packages first and pip packages later. As a result, the dependency resolution in pip happens after conda. As a result it might happen, that constraints on versions introduced in conda are ignored by pip to meet pips dependency resolution. So it makes sense to install everything accessible from conda using conda first to avoid failing dependency resolutions.